internal/clients/config/base.go

```diff
@@ -0,0 +1,97 @@
+package config
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type baseConfig struct {
```

Comment:
Does it make to rename it to `EsBaseConfig`?

Reply:
There's nothing ES specific within this struct though.